File: 2023/ts/01/part01.ts
import fs from "fs";
import path from "path";

// const text = `1abc2
// pqr3stu8vwx
// a1b2c3d4e5f
// treb7uchet
// `;

const text = fs.readFileSync(path.join(__dirname, "./part01.txt"), {
  encoding: "utf-8",
});

const RE = /\d/g;

const sum = text
  .trim()
  .split("\n")
  .map((s) => {
    const ns = s.match(RE)!;
    console.log(ns[0]! + (ns.at(-1) ?? ns[0]!));
    return Number(ns[0]! + (ns.at(-1) ?? ns[0]!));
  })
  .reduce((acc, v) => acc + v, 0);

console.log(sum); // 56049

File: 2023/python/01/part01.py
import re
from pathlib import Path

# text = """1abc2
# pqr3stu8vwx
# a1b2c3d4e5f
# treb7uchet
# """

with open(Path(__file__).parent / "input.txt", "r") as f:
    text = f.read()

RE = re.compile(r"[0-9]")

sm = 0
for s in text.strip().splitlines():
    m = RE.findall(s)
    sm += int(m[0] + m[-1])

print(sm)  # 56049
